### DISCUSSION TITLE
Graphcache query cache miss

### DISCUSSION THREAD

**Discussion by nephix** [urql-graphql/urql#2788] [Q&A]
Hey,

I'm having trouble to understand why the below component and query results in a cache miss.

- urql client uses the `@urql/exchange-graphcache` exchange
- The `useGetUsersQuery` is a graphql query hook generated with GCG.
- The `flag` is just local state causing the component to rerender every second.

The console output of the component is always:

> UsersQuery cache outcome: miss

But in the network tab I don't see any requests when the component rerenders, so it should be a cache hit?

```typescript
const MyComponent: FunctionComponent = () => {
  const [{ data, operation, fetching }] = useGetUsersQuery({
    variables: { ... },
  });
  const [flag, setFlag] = useState(false);

  useEffect(() => {
    setTimeout(() => {
      setFlag(!flag);
    }, 1000);
  }, []);

  console.log(
    'UsersQuery cache outcome:',
    operation?.context.meta?.cacheOutcome,
  );

  return (
    <>
      {fetching && <p>Loading...</p>}

      {!fetching && (
        <Users users={data?.someNamespace?.users.objects} />
      )}
    </>
  );
};
```

I also have a mutation updater which adds users to the list after the mutation ran:

```typescript
export const createUser: GraphCacheUpdaters['Mutation'] = {
  createUser: (result, args, cache) => {
    cache.updateQuery<GetUserListQuery, GetUserListQueryVariables>(
      {
        query: GetUserListDocument,
        variables: {
          ...
        },
      },
      (data): GetUserListQuery | null => {
        if (!data) return null;

        data?.someNamespace?.users.objects?.push(result.createUser.user)
        
        return data;
      },
    );
  },
};
```

When I trigger the mutation, I can see that the updater runs and adds the user to `data`. Afterwards I can see that `MyComponent` rerenders, but 

1. `data` is not the updated data (without the added user from the mutation updater)
2. Keep getting `UsersQuery cache outcome: miss` although there is clearly data in the cache

Is there something I'm missing that causes Graphcache not to pick up and cache the query correctly?
Has it something to do with the query being behind a namespace in the schema and not in the "QueryRoot"?

```graphql
query GetUserList( ... ) {
  someNamespace {
    users( ... ) {
      objects {
        id
      }
    }
  }
}
```

Query result:
```json
{
    "data": {
        "__typename": "QueryRoot",
        "someNamespace": {
            "__typename": "SomeNamespaceQueryRoot",
            "users": {
                "__typename": "UserListOutput",
                "objects": [...]
            }
        }
    },
    "errors": null
}
```

**Comment by Mookiies** [comment]
Are you seeing a network request initially (ie. after a refresh)? Re-rendering the component won't re-trigger the query, it'll just run through that `console.log` with the data that was there originally.

Is your query and updater getting called with identical variables? With different variables the query may not receive the update.

**Reply by nephix** [reply]
> Are you seeing a network request initially

Yes I can see the initial request

> Re-rendering the component won't re-trigger the query, it'll just run through that console.log with the data that was there originally

But then it shouldn't be a cache `miss`?

> Is your query and updater getting called with identical variables

Yes they are same
